client/src/pages/Dashboard/components/AddConversationModal.js

```diff
@@ -0,0 +1,106 @@
+import { useState, forwardRef, useRef } from 'react';
+import { Paper, Grid, Typography, Button, CircularProgress } from '@material-ui/core';
+import { makeStyles } from "@material-ui/core/styles";
+import SearchInput from '../../../components/SearchInput';
+import ConversationItem from './ConversationItem';
+import _ from "lodash";
+import ConversationService from '../../../services/conversationService';
+import { useAuth } from '../../../hooks/useAuth';
+
+const useStyles = makeStyles(theme => ({
+    root: {
+        background: 'white',
+        position: 'absolute',
+        left: "50%",
+        transform: "translate(-50%, 0)",
+        top: "10%",
+        padding: theme.spacing(2.5),
+        width: 500,
+        maxWidth: "90%",
+        maxHeight: "80%",
+        overflow: "auto"
+    },
+    title: {
+        marginBottom: 15,
+    },
+    list: {
+        paddingTop: 0,
+        height: "100%",
+    }
+}));
+
+function AddConversationModal({ addConversationCallback }) {
+    const classes = useStyles();
+    const [users, setUsers] = useState([]);
+    const [isSearching, setIsSearching] = useState(false);
+    const searchRef = useRef(_.debounce(value => updateSearch(value), 300), []).current; // To prevent spamming api requests when typing
+    const auth = useAuth();
+
+    async function updateSearch(value) {
+        if (!value) {
+            setUsers([]);
+        }
+        else {
+            let users = await ConversationService.searchUsers(value);
+            // filter out yourself
+            users = users?.filter(user => user.id !== auth.user.id);
+            setUsers(users);
+        }
+        setIsSearching(false);
+    }
+
+    function handleSearchChanged(event) {
+        if (!event.target.value) {
+            searchRef('');
+            setUsers([]);
```

Comment:
Doesn't `updateSearch` already take care of setting the users to an empty array and setting `isSearching` to false?

Reply:
fixed